### DISCUSSION TITLE
Is supabase API useless because it is lacking proper fascility to get property of embedded tables, autocompletion, poor documentation regarding fetching embedded column with auto completion?

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#5039] [Questions] [answered]
**Fetching properties of foreign key tables is common case in Postgres,** but Supabase lacks this fascility properly. By saying **properly** i mean to say that 
1.  you can do it with some workaround but then there is **_no detail documentation and autocompletion_**  for quering embedded tables. 
2.  It looks more helpful to write sql in supabase dashboad and call it through rpc. Because in sql even for complex query we have a little bit autcompletion working and you can make any kind of query. 
3.  Also **call function through RPC makes client code clean** instead of writing complex query for fetching embedded tables client side **without autocompletion** 

So Why should i use Supabase API other than RPC?

**Comment by esDotDev** [comment, marked as answer]
So far for me, the lack of static typing on the queries has been mitigated quite a bit by the excellent error reporting. If you mis-type any column names, it immediately tells you what you did wrong. If it just swallowed bad queries it would be a much bigger issues.
